server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -582,21 +595,47 @@ private boolean isExpired(Subscription subscription) {
     }
 
     @Transactional
-    private void deleteExcessEntitlements(Pool existingPool) {
+    private void deleteExcessEntitlements(List<Pool> existingPools) {
         boolean lifo = !config
             .getBoolean(ConfigProperties.REVOKE_ENTITLEMENT_IN_FIFO_ORDER);
+        if (existingPools != null && !existingPools.isEmpty()) {
```

Comment:
Rather than nesting basically the entire method in a negated condition, why not

```
if (existingPools == null || existingPools.isEmpty()) {
    return;
}
// Actual logic here
```